eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version1/ChannelCodecs1.scala

```diff
@@ -0,0 +1,266 @@
+package fr.acinq.eclair.wire.internal.channel.version1
+
+import fr.acinq.bitcoin.DeterministicWallet.{ExtendedPrivateKey, KeyPath}
+import fr.acinq.bitcoin.{ByteVector32, OutPoint, Transaction, TxOut}
+import fr.acinq.eclair.MilliSatoshi
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.crypto.ShaChain
+import fr.acinq.eclair.transactions.Transactions._
+import fr.acinq.eclair.transactions.{CommitmentSpec, DirectedHtlc, IncomingHtlc, OutgoingHtlc}
+import fr.acinq.eclair.wire.CommonCodecs.{bytes32, bytes64, mapCodec, publicKey, setCodec, varintoverflow, _}
+import fr.acinq.eclair.wire.LightningMessageCodecs.{channelAnnouncementCodec, channelUpdateCodec, closingSignedCodec, commitSigCodec, fundingCreatedCodec, fundingLockedCodec, fundingSignedCodec, lightningMessageCodec, shutdownCodec, updateAddHtlcCodec, _}
+import fr.acinq.eclair.wire.UpdateMessage
+import scodec.codecs.{bits, bool, byteAligned, bytes, discriminated, either, int64, listOfN, optional, provide, uint16, uint32, variableSizeBytesLong, _}
+import scodec.{Attempt, Codec}
+
+private[channel] object ChannelCodecs1 {
+
+  private[version1] object Codecs {
+
+    /**
+     * All LN protocol message must be stored as length-delimited, because they may have arbitrary trailing data
+     */
+    def lengthDelimited[T](codec: Codec[T]): Codec[T] = variableSizeBytesLong(varintoverflow, codec)
```

Comment:
Shouldn't this one be in the `CommonCodecs`?

Reply:
Yes this one could, but only that one I think.

Reply:
Done with ee9050d46.